### DISCUSSION TITLE
How to build customize RAC Combobox to conditionally render a Section or ListBoxItem

### DISCUSSION THREAD

**Discussion by camoy1** [adobe/react-spectrum#6822] [Q&A] [answered]
I want to create a reusable wrapper around RAC `Combobox` that conditionally renders the items appropriately based on whether the item type is a section and render the items using the RAC `Section`, `Header`, and `Collection` components, otherwise the item is rendered as ListBoxItem. I found [this example](https://codesandbox.io/p/sandbox/romantic-chaum-j88ng3?file=%2Fsrc%2FListBox.tsx%3A91%2C5-102%2C10) using react-aria and react-stately hooks; however, is there a way to build this type of component using the `react-aria-components` instead of building the scratch using the hooks?

For example, based on the ComboBox docs's [reusable wrapper](https://react-spectrum.adobe.com/react-aria/ComboBox.html#reusable-wrappers) – where the children are rendered inside the `ListBox` is where I'd check if the state collection's type is _section_ and render the appropriate components, otherwise it'd render the `ListBoxItem`.

**Comment by reidbarber** [comment, marked as answer]
I would recommend looking at this example: https://react-spectrum.adobe.com/react-aria/ComboBox.html#dynamic-items

But instead of making the assumption that the top level is a section like this example does, you'll want to check the item for what signifies a section (maybe you have `type` in your items, or you check for `children`), then conditionally render Section or ListBoxItem based on that.
